src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -93,32 +93,39 @@ internal static bool DoSslHandshake(SafeSslHandle context, IntPtr recvPtr, int r
 
             if (sendCount > 0)
             {
-                sendPtr = Marshal.AllocHGlobal(sendCount);
+                sendBuf = new byte[sendCount];
 
                 try
                 {
-                    sendCount = BioRead(context.OutputBio, sendPtr, sendCount);
+                    sendCount = BioRead(context.OutputBio, sendBuf, sendCount);
                 }
                 finally
                 {
                     if (sendCount <= 0)
                     {
-                        Marshal.FreeHGlobal(sendPtr);
-                        sendPtr = IntPtr.Zero;
+                        sendBuf = null;
                         sendCount = 0;
                     }
                 }
             }
-        
+
             return ((libssl.SSL_state(context) == (int)libssl.SslState.SSL_ST_OK));
 
         }
 
-        internal static int Encrypt(SafeSslHandle context, IntPtr buffer, int offset, int count, int bufferCapacity, out libssl.SslErrorCode errorCode)
+        internal static int Encrypt(SafeSslHandle context, byte[] buffer, int offset, int count, int bufferCapacity, out libssl.SslErrorCode errorCode)
         {
             errorCode = libssl.SslErrorCode.SSL_ERROR_NONE;
 
-            int retVal = libssl.SSL_write(context, new IntPtr(buffer.ToInt64() + offset), count);
+            int retVal;
```

Comment:
Preexisting your change, but especially now that we're using unsafe code, we should consider adding some Debug.Asserts here to validate that we're getting expected inputs for buffer/offset/count.